### PR TITLE
[[ Bug 15612 ]] Reinstate MCtodestroy stack list (7.0)

### PR DESCRIPTION
**Pull request by runrevmark**
7.0 version of #2626

### CODE REVIEW

**Comment by livecodesebastien** [comment]
@livecode-vulcan review ok  870e5c8

**Comment by livecode-vulcan** [comment]
💙 review by @livecodesebastien ok 870e5c861a34f4755241fd0eff8cb190ae891043

**Comment by livecode-vulcan** [comment]
😞 test failure 870e5c861a34f4755241fd0eff8cb190ae891043
- try-linux-x86: [pending](https://vulcan.livecode.com/builders/try-linux-x86/builds/624)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/684)
- try-ios: [failure](https://vulcan.livecode.com/builders/try-ios/builds/573)
- try-android-armv6: [pending](https://vulcan.livecode.com/builders/try-android-armv6/builds/637)
- try-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/569)
- try-mac: [pending](https://vulcan.livecode.com/builders/try-mac/builds/562)

File: engine/src/sellst.cpp [hunk 1]
```diff
@@ -423,7 +423,11 @@ bool MCSellist::clipboard(bool p_is_cut)
 					if (tptr -> ref -> getstack() -> iskeyed())
 					{
 						if (tptr -> ref -> del())
+                        {
+                            if (tptr -> ref -> gettype() == CT_STACK)
+                                MCtodestroy -> remove(static_cast<MCStack *>(tptr -> ref))
```
Comment (livecodesebastien) [inline engine/src/sellst.cpp]: Missing a ';'

**Comment by livecodesebastien** [comment]
@livecode-vulcan review ok 35dda23

**Comment by livecode-vulcan** [comment]
💙 review by @livecodesebastien ok 35dda23b21a54f2ca369cda8174cb5b8df30864c

**Comment by livecode-vulcan** [comment]
😞 test failure 35dda23b21a54f2ca369cda8174cb5b8df30864c
- try-linux-x86: [failure](https://vulcan.livecode.com/builders/try-linux-x86/builds/625)
- try-win-x86: [pending](https://vulcan.livecode.com/builders/try-win-x86/builds/685)
- try-ios: [failure](https://vulcan.livecode.com/builders/try-ios/builds/574)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/638)
- try-linux-x86_64: [failure](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/570)
- try-mac: [failure](https://vulcan.livecode.com/builders/try-mac/builds/563)

**Comment by livecodesebastien** [comment]
@livecode-vulcan review not ok 35dda23

File: engine/src/exec-pasteboard.cpp [hunk 2]
```diff
@@ -457,7 +457,11 @@ void MCPasteboardProcessToClipboard(MCExecContext& ctxt, MCObjectPtr *p_targets,
 			for(uint4 i = 0; i < p_object_count; ++i)
 			{
 				if (p_targets[i] . object -> del())
+                {
+                    if (p_targets[i] . object -> gettype() == CT_STACK)
+                        MCtodestroy -> remove(static_cast<MCStack *>(_targets[i] . object));
```
Comment (livecodesebastien) [inline engine/src/exec-pasteboard.cpp]: Should be `p_targets`

**Comment by livecode-vulcan** [comment]
💔 review by @livecodesebastien not ok 35dda23b21a54f2ca369cda8174cb5b8df30864c

**Comment by livecodesebastien** [comment]
@livecode-vulcan review ok d5b957b

**Comment by livecode-vulcan** [comment]
💙 review by @livecodesebastien ok d5b957bc8b4764d4c9dc73813be5eeeef5bc0896

**Comment by livecode-vulcan** [comment]
😎 test success d5b957bc8b4764d4c9dc73813be5eeeef5bc0896
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/675)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/736)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/624)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/688)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/620)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/613)

**Pull request merged**